lib/fog/libvirt/models/compute/server.rb

```diff
@@ -97,6 +97,11 @@ def destroy(options={ :destroy_volumes => false, :flags => ::Libvirt::Domain::UN
           if flags.zero?
             service.vm_action(uuid, :undefine)
           else
+            # the test driver doesn't support UNDEFINE_NVRAM and we can't probe
+            # for the capability at runtime
+            if service.uri.uri == 'test:///default'
```

Comment:
I did some further digging. https://libvirt.org/drvtest.html documents the test driver. So really it should look at the scheme see if that's `test` or `test+something`.